static/app/views/performance/trends/changeExplorerUtils/metricsChart.tsx

```diff
@@ -0,0 +1,468 @@
+    eventView: beforeEventView,
+    location: beforeLocation,
+    orgSlug: organization.slug,
+    transactionName: transaction.transaction,
+    transactionThresholdMetric: TransactionThresholdMetric.TRANSACTION_DURATION,
+    options: {
+      refetchOnWindowFocus: false,
+    },
+  });
+
+  const {data: afterBreakpoint, isLoading: isLoadingAfter} = useDiscoverQuery({
+    eventView: afterEventView,
+    location: afterLocation,
+    orgSlug: organization.slug,
+    transactionName: transaction.transaction,
+    transactionThresholdMetric: TransactionThresholdMetric.TRANSACTION_DURATION,
+    options: {
+      refetchOnWindowFocus: false,
+    },
+  });
+
+  const {data: beforeBreakpointErrors, isLoading: isLoadingBeforeErrors} =
+    useDiscoverQuery({
+      eventView: beforeErrorsEventView,
+      location: beforeErrorsLocation,
+      orgSlug: organization.slug,
+      transactionName: transaction.transaction,
+      transactionThresholdMetric: TransactionThresholdMetric.TRANSACTION_DURATION,
+      options: {
+        refetchOnWindowFocus: false,
+      },
+    });
+
+  const {data: afterBreakpointErrors, isLoading: isLoadingAfterErrors} = useDiscoverQuery(
+    {
+      eventView: afterErrorsEventView,
+      location: afterErrorsLocation,
+      orgSlug: organization.slug,
+      transactionName: transaction.transaction,
+      transactionThresholdMetric: TransactionThresholdMetric.TRANSACTION_DURATION,
+      options: {
+        refetchOnWindowFocus: false,
+      },
+    }
+  );
+
+  const throughput: TableDataRow[] = getEventsRowData(
+    'tps()',
+    'Throughput',
+    'ps',
+    '-',
+    false,
+    beforeBreakpoint,
+    afterBreakpoint
+  );
+
+  fieldsNeeded.includes('p50') &&
+    (p50 = getEventsRowData(
+      'p50()',
+      'P50',
+      'ms',
+      '-',
+      false,
+      beforeBreakpoint,
+      afterBreakpoint
+    ));
+
+  fieldsNeeded.includes('p95') &&
+    (p95 = getEventsRowData(
+      'p95()',
+      'P95',
+      'ms',
+      '-',
+      false,
+      beforeBreakpoint,
+      afterBreakpoint
+    ));
+
+  const errors: TableDataRow[] = getEventsRowData(
+    'count()',
+    'Errors',
+    '',
+    0,
+    true,
+    beforeBreakpointErrors,
+    afterBreakpointErrors
+  );
+
+  const columnOrder = MetricColumnOrder.map(column => COLUMNS[column]);
+
+  return (
+    <GridEditable
+      data={[...throughput, ...p50, ...p95, ...errors]}
+      columnOrder={columnOrder}
+      columnSortBy={[]}
+      grid={{
+        renderHeadCell,
+        renderBodyCell,
+      }}
+      location={location}
+      isLoading={
+        isLoadingBefore ||
+        isLoadingAfter ||
+        isLoading ||
+        isLoadingBeforeErrors ||
+        isLoadingAfterErrors
+      }
+    />
+  );
+}
+
+function getEventsRowData(
+  field: string,
+  rowTitle: string,
+  suffix: string,
+  nullValue: string | number,
+  wholeNumbers: boolean,
+  beforeData?: TableData,
+  afterData?: TableData
+): TableDataRow[] {
```

Comment:
Nit: I don't think this needs to return an array, it can just return one object or another. You're expanding all these 1-item arrays anyway.